api/objects.proto

```diff
@@ -267,3 +268,50 @@ message Secret {
 	// Whether the secret is an internal secret (not set by a user) or not.
 	bool internal = 4;
 }
+
+// Resource is a top-level object with externally defined content and indexing.
+// SwarmKit can serve as a store for these objects without understanding their
+// meanings.
+message Resource {
```

Comment:
@aluzzardi says he doesn't like `Record`, because it sounds very DB-like. He prefers `Object`, though this is over-generic.